src/UI/Component/Triggerer.php

```diff
@@ -0,0 +1,33 @@
+<?php
+namespace ILIAS\UI\Component;
+
+/**
+ * Interface Triggerer
+ *
+ * Describes a component that can trigger signals of other components on given events, such as click or hover.
+ * All supported events are abstracted with interfaces (see Clickable, Hoverable).
+ * Example: A button can trigger the show signal of a modal on click (which will open the modal on button click).
+ *
+ * @package ILIAS\UI\Component
+ */
+interface Triggerer {
+
+	const EVENT_CLICK = 'click';
+	const EVENT_HOVER = 'hover';
+	const EVENT_ONLOAD = 'ready';
```

Comment:
Those three constants seem awkward. Why are they placed here. I think the base interface should be agnostic of it's derivates. For some reason, they are wired up here by constants.